tensorflow_lite_support/c/test/task/vision/image_classifier_test.cc

```diff
@@ -42,51 +42,88 @@ constexpr char kMobileNetQuantizedWithMetadata[] =
     "mobilenet_v1_0.25_224_quant.tflite";
 
 StatusOr<ImageData> LoadImage(const char* image_name) {
-  return DecodeImageFromFile(JoinPath("./" /*test src dir*/,
-                                      kTestDataDirectory, image_name));
+  return DecodeImageFromFile(
+      JoinPath("./" /*test src dir*/, kTestDataDirectory, image_name));
 }
 
 class ImageClassifierFromOptionsTest : public tflite_shims::testing::Test {};
 
+TEST_F(ImageClassifierFromOptionsTest, FailsWithNullOptionsAndError) {
+  TfLiteSupportError* error = nullptr;
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(nullptr, &error);
+
+  EXPECT_EQ(image_classifier, nullptr);
+  if (image_classifier) TfLiteImageClassifierDelete(image_classifier);
```

Comment:
nit: line 57 is redundant. image_classifier should just be a nullptr because of line 56.

Reply:
But `EXPECT_EQ` will not stop execution on failure. EXPECT_EQ fails when `image_classifier` is not null. Logically, it will have to be deleted after that right? Hence the null check too.
But I do agree that in reality, the delete will never be executed in this particular test as the image classifier would not get created unless the C or C++ layer makes the mistake of creating some default options with a valid ExternalFile if options parameter is null. Chances of that happening are very less.
But I still put it in for the sake of code correctness. LMK if it needs to be removed.

Reply:
I see. I would change `EXPECT_EQ(image_classifier, nullptr);` => `ASSERT_EQ(image_classifier, nullptr);`. But it's up to you.

Reply:
But if this image classifier case fails but error is still non null, it won't be deleted if we change to `ASSERT` right, since error checks and cleanup are after this `EXPECT`

Reply:
No, it won't. That's an alternative if you don't want to delete line 57, but  I'm OK with either way.